Lib/unittest/main.py

```diff
@@ -225,8 +241,7 @@ def _do_discovery(self, argv, Loader=None):
                 self._initArgParsers()
             self._discovery_parser.parse_args(argv, self)
 
-        loader = self.testLoader if Loader is None else Loader()
-        self.test = loader.discover(self.start, self.pattern, self.top)
+        self.createTests(from_discovery=True, Loader=Loader)
```

Comment:
I'm curious, why was this change necessary?

Reply:
This refactoring was necessary because I didn't want to repeat the code that sets `loader.testNamePatterns` from the CLI options.  That code has to be executed BEFORE `loader.discover`. Besides, I believe it makes sense to have a single place where `self.test` is set.

Reply:
This refactoring was necessary because I didn't want to repeat the code that sets `loader.testNamePatterns` from the CLI options.  That code has to be executed BEFORE `loader.discover`. Besides, I believe it makes sense to have a single place where `self.test` is set.